openff/toolkit/utils/rdkit_wrapper.py

```diff
@@ -507,23 +974,30 @@ def _fuzzy_query(query):
         # N.B. This isn't likely to be an active
         generic_mol = (
             Chem.MolFromSmarts(  # TODO: optimisation, create this once somewhere
-                "".join("[#{}]".format(i + 1) for i in range(112))
+                "[*]" + "".join("[#{}]".format(i + 1) for i in range(112))
             )
         )
 
         fuzzy = Chem.Mol(query)
-        for a in fuzzy.GetAtoms():
+        neighbor_idxs = []
+        for idx, a in enumerate(fuzzy.GetAtoms()):
             a.SetFormalCharge(0)
-            a.SetQuery(
-                generic_mol.GetAtomWithIdx(a.GetAtomicNum() - 1)
-            )  # i.e. H looks up atom 0 in our generic mol
+            if strict_degree and a.GetAtomicNum() > 0:
+                a.SetQuery(
+                    Chem.AtomFromSmarts(f"[#{a.GetAtomicNum()}D{a.GetDegree()}]")
+                )
+            else:
+                a.SetQuery(
+                    generic_mol.GetAtomWithIdx(a.GetAtomicNum())
+                )  # i.e. H looks up atom 0 in our generic mol
```

Comment:
This is no longer true
```suggestion
                )  
```